src/js/controllers/walletHome.js

```diff
@@ -1225,44 +1226,81 @@ angular.module('copayApp.controllers').controller('walletHomeController', functi
     return actions.hasOwnProperty('create');
   };
 
-  this._doSendAll = function(amount) {
+  this._doSendMax = function(amount) {
     this.setForm(null, amount, null);
   };
 
-  this.sendAll = function(totalBytesToSendMax, availableBalanceSat) {
+  this.sendMax = function(totalBytesToSendMax, availableBalanceSat) {
     var self = this;
+    var fc = profileService.focusedClient;
     var availableMaxBalance;
-    var feeToSendMaxStr;
+    var feeToSendMaxSat;
     this.error = null;
+    this.warning = null;
     this.setOngoingProcess(gettextCatalog.getString('Calculating fee'));
 
     feeService.getCurrentFeeValue(function(err, feePerKb) {
-      self.setOngoingProcess();
-      if (err || lodash.isNull(feePerKb)) {
+      if (err || !lodash.isNumber(feePerKb)) {
+        self.setOngoingProcess();
         self.error = gettext('Could not get fee value');
         return;
       }
 
-      var feeToSendMaxSat = parseInt(((totalBytesToSendMax * feePerKb) / 1000.).toFixed(0));
-      if (availableBalanceSat > feeToSendMaxSat) {
-        self.lockedCurrentFeePerKb = feePerKb;
-        availableMaxBalance = strip((availableBalanceSat - feeToSendMaxSat) * self.satToUnit);
-        feeToSendMaxStr = profileService.formatAmount(feeToSendMaxSat) + ' ' + self.unitName;
-      } else {
-        self.error = gettext('Not enought funds for fee');
-        return;
-      }
+      feeToSendMaxSat = parseInt(((totalBytesToSendMax * feePerKb) / 1000.).toFixed(0));
+      availableMaxBalance = strip((availableBalanceSat - feeToSendMaxSat) * self.satToUnit);
 
-      var msg = gettextCatalog.getString("{{fee}} will be deducted for bitcoin networking fees", {
-        fee: feeToSendMaxStr
-      });
+      var opts = {};
+      opts.feePerKb = feePerKb;
+      opts.returnInputs = true;
+      var config = configService.getSync();
+      opts.excludeUnconfirmedUtxos = !config.wallet.spendUnconfirmed;
+      self.setOngoingProcess(gettextCatalog.getString('Retrieving inputs information'));
 
-      $scope.$apply();
-      confirmDialog.show(msg, function(confirmed) {
-        if (confirmed) {
-          self._doSendAll(availableMaxBalance);
-        } else {
-          self.resetForm();
+      fc.getSendMaxInfo(opts, function(err, resp) {
+        if (err) {
+          self.setOngoingProcess();
+          self.error = err;
+          $scope.$apply();
+          return;
+        }
+        if (resp.amount == 0) {
+          self.setOngoingProcess();
+          self.error = gettext("Not enought funds for fee");
+          $scope.$apply();
+          return;
+        }
+        if (availableMaxBalance == 0) {
```

Comment:
This check can be done before calling `getSendMaxInfo()` and should be performed before checking `resp.amount`.